hack/e2e/ensure_htpasswd.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# Check if htpasswd is installed and install it if not
+verify_htpasswd_version()
```

Comment:
We're not actually verifying htpasswd version, but just its existence, so this function name is misleading.

```suggestion
verify_htpasswd()
```